src/sentry/api/endpoints/integrations/sentry_apps/rotate_secret.py

```diff
@@ -0,0 +1,62 @@
+from django.http import Http404
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import control_silo_endpoint
+from sentry.api.bases.sentryapps import SentryAppBaseEndpoint
+from sentry.api.permissions import SentryPermission
+from sentry.api.serializers import serialize
+from sentry.models.apiapplication import generate_token
+from sentry.models.integrations.sentry_app import SentryApp
+from sentry.services.hybrid_cloud.organization import organization_service
+from sentry.services.hybrid_cloud.user.service import user_service
+
+
+class SentryAppRotateSecretPermission(SentryPermission):
+    scope_map = {
+        "POST": ["org:write", "org:admin"],
+    }
+
+    def has_object_permission(self, request: Request, view: object, sentry_app: SentryApp):
+        # organization that owns an integration
+        org_context = organization_service.get_organization_by_id(id=sentry_app.owner_id)
+        if org_context is None:
+            raise Http404
+
+        organization = org_context.organization
+        if organization is None:
+            raise Http404
+
+        # if user is not a member of an organization owning an integration,
+        # return 404 to avoid leaking integration slug
+        organizations = (
+            user_service.get_organizations(user_id=request.user.id)
+            if request.user.id is not None
+            else ()
+        )
+        if not any(organization.id == org.id for org in organizations):
+            raise Http404
+
+        # permission check inside an organization
+        self.determine_access(request, organization)
+        allowed_scopes = set(self.scope_map.get(request.method or "", []))
+        return any(request.access.has_scope(s) for s in allowed_scopes)
```

Comment:
Would it make more sense to just return false/boolean and have the parent/calling method decide what error code it should use? I think instead of 404, a failed permission is more of a 403?

Reply:
I think maybe even a `400` bad request status code might be more accurate? That being said, do we think this is plausible? Can a SentryApp (id or object) exist without a representational organization?

Reply:
Because technically a 404 would mean the sentry_app itself doesn't exist, which isn't really true in this case right?

Reply:
Technically 403 means failed permission, but in this case we don't want to leak info if a specific integration slug exists with different status codes. Take a look from an attacker's perspective:
```
POST /integration-aaa/rotate-secret/  404 -- integration does not exist
POST /integration-bb/rotate-secret/   403 -- aha, so it exists but I don't have permissions. I can try to build some attack on top of this info
```
We're taking a similar approach in other places, for example:
https://github.com/getsentry/sentry/blob/master/src/sentry/api/bases/sentryapps.py#L222-L224

Reply:
Ahhh, I see what you're saying, that makes sense!

Reply:
I doubt we are, but hope we're that consistent across our other API endpoints in terms of behavior and expected status codes